db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -3044,6 +3059,7 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
     m->in_progress = false;  // not being processed anymore
   }
   TEST_SYNC_POINT("DBImpl::BackgroundCompaction:Finish");
+  io_s.PermitUncheckedError();
```

Comment:
Can you move this line to earlier?  It seems like it belongs around 2943 (if status.ok() && ! io_s.ok()))

Reply:
Sure.